src/posthog-featureflags.ts

```diff
@@ -177,32 +178,39 @@ export class PostHogFeatureFlags {
         const token = this.instance.config.token
         const personProperties = this.instance.get_property(STORED_PERSON_PROPERTIES_KEY)
         const groupProperties = this.instance.get_property(STORED_GROUP_PROPERTIES_KEY)
-        const json_data = JSON.stringify({
+        const json_data = {
             token: token,
             distinct_id: this.instance.get_distinct_id(),
             groups: this.instance.getGroups(),
             $anon_distinct_id: this.$anon_distinct_id,
             person_properties: personProperties,
             group_properties: groupProperties,
             disable_flags: this.instance.config.advanced_disable_feature_flags || undefined,
-        })
+        }
+
+        this.instance._send_request({
+            method: 'POST',
+            url: this.instance.requestRouter.endpointFor('api', '/decide/?v=3'),
+            data: json_data,
+            compression: Compression.Base64,
+            timeout: this.instance.config.feature_flag_request_timeout_ms,
+            callback: (response) => {
+                this.setReloadingPaused(false)
 
-        const encoded_data = _base64Encode(json_data)
-        this.instance._send_request(
-            this.instance.requestRouter.endpointFor('api', '/decide/?v=3'),
-            { data: encoded_data },
-            { method: 'POST' },
-            (response) => {
+                if (response.statusCode !== 200) {
+                    return // or error out??
+                }
                 // reset anon_distinct_id after at least a single request with it
                 // makes it through
                 this.$anon_distinct_id = undefined
-                this.receivedFeatureFlags(response as DecideResponse)
+                if (response.json) {
```

Comment:
Hmm, this is a bit problematic compared to old behaviour, since this blocks anti-flickering behaviour from ever returning if flags fail to load.

I'll push a change to allow this, and write some tests for this behaviour, seems to be undocumented.